hw-event-proxy/cmd/main.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2021 The Cloud Native Events Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	hwevent "github.com/redhat-cne/sdk-go/pkg/hwevent"
+
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/pb"
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/restclient"
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/util"
+	"google.golang.org/grpc"
+
+	v1hwevent "github.com/redhat-cne/sdk-go/v1/hwevent"
+	v1pubsub "github.com/redhat-cne/sdk-go/v1/pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	hwEventVersion string = "v1"
+	eventType      string = "HW_EVENT"
+)
+
+var (
+	apiPath         = "/api/cloudNotifications/v1/"
+	apiPort         int
+	json            = jsoniter.ConfigCompatibleWithStandardLibrary
+	pub             pubsub.PubSub
+	resourceAddress string
+)
+
+func main() {
+	flag.IntVar(&apiPort, "api-port", 8080, "The address the rest api endpoint binds to.")
+	flag.Parse()
+	util.InitLogger()
+
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		log.Error("cannot find NODE_NAME environment variable,setting to default `mock` node")
+		nodeName = "mock"
+	}
+
+	resourceAddress = fmt.Sprintf("/cluster/node/%s/redfish/event", nodeName)
+
+	var err error
+	pub, err = createPublisher()
+	if err != nil {
+		return
```

Comment:
is this os.Exit(1)?

Reply:
meaning , if publish create fails do you retry or just pod errors out ?. The sidecar rest service may not be ready before you could make  rest service. You need to check for avialability and then make this call.

Reply:
added sidecar api health check at line 75

Reply:
the return statement returns from the  main function , this ends the pod.

Reply:
added retry for creating publisher

Reply:
I am closing this for now. But if publisher fails it should be os.Exit(1)